spec/restaurant_spec.rb

```diff
@@ -0,0 +1,37 @@
+require './lib/restaurant.rb'
+
+describe Restaurant do
+
+  let(:order) {double :order, empty?: false}
+  subject(:subject) { Restaurant.new(order) }
+
+  describe "#initialization" do
+
+    it "is initialized with an order" do
+      expect(subject).to respond_to :order
+    end
+  end
+
+  it "checks if given total equals to order's total" do
+    allow(order).to receive(:total).and_return(35)
+    expect(subject.total_correct(35)).to eq true
+  end
+
+  it "knows the list of ordered dishes" do
+    allow(order).to receive(:basket).and_return({:coke => 1})
```

Comment:
Style/BracesAroundHashParameters: Redundant curly braces around a hash parameter.<br>Style/HashSyntax: Use the new Ruby 1.9 hash syntax. (https://github.com/bbatsov/ruby-style-guide#hash-literals)